src/main/java/seedu/jarvis/ui/CommandBox.java

```diff
@@ -37,13 +53,63 @@ public CommandBox(CommandExecutor commandExecutor) {
     @FXML
     private void handleCommandEntered() {
         try {
-            commandExecutor.execute(commandTextField.getText());
+            String newUserInput = commandTextField.getText();
+            //Storing the command
+            historyCommands.add(newUserInput);
+            this.inputPointer = historyCommands.size();
+
+            //Execute the command
+            commandExecutor.execute(newUserInput);
             commandTextField.setText("");
+
         } catch (CommandException | ParseException e) {
             setStyleToIndicateCommandFailure();
         }
     }
 
+    /**
+     * Handles the UP and DOWN KeyCode pressed events.
+     */
+    @FXML
+    private void handleKeyEvent() {
+        commandTextField.setOnKeyPressed(event -> {
+            if (event.getCode() == KeyCode.UP && inputPointer > 0) {
+                goToPreviousInput();
+
+            } else if (event.getCode() == KeyCode.DOWN && inputPointer < historyCommands.size()) {
+                goToNextInput();
+            }
+        });
+    }
+
+    /**
+     * Goes to the previous input.
+     */
+    private void goToPreviousInput() {
+        logger.info("===== CommandBox detected UP key: Previous user input accessed.");
```

Comment:
The amount of logging is great, especially now when we are trying to reduce bugs